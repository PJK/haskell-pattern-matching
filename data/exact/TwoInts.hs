f :: Int -> Int
f 0 = 1
f 1 = 2
