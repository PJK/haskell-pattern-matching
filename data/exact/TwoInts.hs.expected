{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "f",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "fresh0"
                            }
                        ],
                        "Constraints:\n  fresh10 == False\n  fresh10 == fresh0 == 1\n  fresh4 == False\n  fresh4 == fresh0 == 0\n\nSatisfiable. Model:\n  fresh10 = False :: Bool\n  fresh0  =     2 :: Integer\n  fresh4  = False :: Bool\n"
                    ]
                ]
            }
        ]
    ]
}
